### PR TITLE
prov/shm: coverity/bug fixes

### PR DESCRIPTION
**Pull request by aingerson**
Fixes a few bugs when processing rma ops:
- If rx_cq is full and remote data is requested, don't discard the commands
- If mr verification failed, discard both cmds but don't process message
- Don't set rx_cq entry buf field when generating completion

Signed-off-by: aingerson <alexia.ingerson@intel.com>

### CODE REVIEW

File: prov/shm/src/smr_ep.c [hunk 1]
```diff
@@ -478,7 +478,7 @@ static int smr_progress_cmd_rma(struct smr_ep *ep, struct smr_cmd *cmd)
 	    ofi_cirque_isfull(ep->util_ep.rx_cq->cirq)) {
 		FI_WARN(&smr_prov, FI_LOG_EP_CTRL,
 			"rx cq full\n");
-		ret = -FI_ENOSPC;
+		return -FI_ENOSPC;
```
Comment (shefty) [inline prov/shm/src/smr_ep.c]: Is this return value ever reported back to the user?

Comment (shefty) [reply prov/shm/src/smr_ep.c]: never mind - I just looked.  If things are working correctly, we should never hit this anyway -- at least I think

Comment (aingerson) [reply prov/shm/src/smr_ep.c]: Yeah, these error codes are only used internally to indicate whether to continue to progress or not and this one I don't think should ever be an issue. I can take it out later if I determine this for sure.
Also, any idea why AppVeyor is failing?

Comment (shefty) [reply prov/shm/src/smr_ep.c]: No idea -- it stopped on 1 of the 6 builds for some reason without a useful error message.  I restarted them.

**Pull request merged**